new_client/src/components/events/AddEvent.jsx

```diff
@@ -0,0 +1,250 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+import '../auth/Error.css';
+
+class AddEvent extends Component {
+  state = {
+    event: {
+      eventName: '',
+      description: '',
+      date: '',
+      venue: '',
+      graduationYear: '',
+      selectedFile: null
+    },
+    error: {
+      eventNameError: '',
+      descriptionError: '',
+      dateError: '',
+      venueError: '',
+      graduationYearError: ''
+    }
+  };
+
+  onFileChange = (e) => {
+    const event = { ...this.state.event };
+    event.selectedFile = e.target.files[0];
+    this.setState({ event: event });
+  };
+
+  handleChange = (e) => {
+    const field = e.target.name;
+
+    const event = { ...this.state.event, [field]: e.target.value };
+
+    this.setState({
+      event: event
+    });
+  };
+
+  isValid = () => {
+    const error = {
+      ...this.state.error
+    };
+    let ret = true;
+    if (this.state.event.eventName === '') {
+      error.eventNameError = '*Event name cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.description === '') {
+      error.descriptionError = '*Event description cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.venue === '') {
+      error.venueError = '*Event venue cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.date === '') {
+      error.dateError = '*Event date cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.graduationYear === '') {
+      error.graduationYearError = '*Graduation year cannot be empty';
+      ret = false;
+    }
+    this.setState({ error: error });
+    return ret;
+  };
+
+  handleSubmit = (event) => {
+    if (this.isValid()) {
+      const formData = new FormData();
+      if (this.state.event.selectedFile) {
+        formData.append(
+          'COC_Event',
+          this.state.event.selectedFile,
+          this.state.event.selectedFile.name
+        );
+      }
+      const {
+        eventName,
+        description,
+        date,
+        venue,
+        graduationYear
+      } = this.state.event;
+      formData.append('eventName', eventName);
+      formData.append('description', description);
+      formData.append('date', date);
+      formData.append('venue', venue);
+      formData.append('graduationYear', graduationYear);
+      if (this.props.isUpdating) {
+        const updatingEventId = this.props.updatingEvent._id;
+        axios
+          .put(
+            process.env.REACT_APP_API + `/events/${updatingEventId}`,
+            formData
+          )
+          .then((res) => {
+            console.log(res.data, this.props.updatingEvent);
+          })
+          .catch((err) => {
+            console.log(err);
+          });

```

Comment:
While logging is useful, a successful/unsuccessful request must be relayed to the user, through an alert perhaps.